[Store][Supabase] Add store

junaidbinfarooq opened the pull request:
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| Docs?         | yes
| Issues        | Fix #320 
| License       | MIT

This PR adds support for Supabase as a vector store backend, leveraging Supabase's pgvector extension through the REST API.

### Features

- **Supabase Store Bridge**: Full integration with Supabase's REST API and pgvector extension
- **Bundle Configuration**: Complete YAML configuration support for the AI Bundle
- **Batch Operations**: Efficient document insertion with chunking (200 docs per request)
- **Vector Search**: Similarity search with configurable match count and score thresholds
- **Manual Setup**: Clear documentation for required database schema setup

### Key Differences from Postgres Store

Unlike the direct Postgres store, the Supabase bridge:
- Uses REST API instead of direct database connections
- Requires manual database setup (tables, functions, indexes)
- Supports Supabase's hosted PostgreSQL with pgvector

junaidbinfarooq commented on examples/rag/supabase.php:
```diff
@@ -0,0 +1,99 @@
+
+require_once dirname(__DIR__).'/bootstrap.php';
+
+echo "Make sure you've run the SQL setup from SUPABASE_SETUP.md first!\n\n";
```
@chr-hertel @OskarStark 
Removed it. Other review comments are also addressed.
Please re-review the PR.

OskarStark commented on examples/rag/supabase.php:
```diff
@@ -0,0 +1,97 @@
+    http_client()
+);
+
+$embeddingModel = new Ollama('mxbai-embed-large');
+$vectorizer = new Vectorizer($platform, $embeddingModel);
```
```suggestion
$vectorizer = new Vectorizer($platform, new Ollama('mxbai-embed-large'));
```

OskarStark commented on examples/rag/supabase.php:
```diff
@@ -0,0 +1,97 @@
+);
+
+echo "Query: Which movie fits the theme of technology?\n";
+echo "Processing...\n";
```
```suggestion
```
Should be that fast, that this is not needed

OskarStark commented on src/ai-bundle/doc/index.rst:
```diff
@@ -109,7 +109,7 @@ Configuration
                 tools: false
         store:
-            # also azure_search, meilisearch, memory, mongodb, pinecone, qdrant and surrealdb are supported as store type
+            # also azure_search, meilisearch, memory, mongodb, pinecone, qdrant, surrealdb, and supabase are supported as store type
```
I removed this comment in 
* https://github.com/symfony/ai/pull/614

Please rebase

OskarStark commented on src/ai-bundle/config/options.php:
```diff
@@ -375,6 +375,19 @@
+                        ->useAttributeAsKey('name')
+                        ->arrayPrototype()
+                            ->children()
+                                ->scalarNode('url')->isRequired()->cannotBeEmpty()->end()
```
Following #604, please make the http client configurable, you can barely copy paste from #604, thanks

junaidbinfarooq replied:
I do not see it done for any other store. Am I missing something here?

OskarStark commented on src/store/src/Bridge/Supabase/Store.php:
```diff
@@ -0,0 +1,155 @@
+ *
+ * @see https://github.com/pgvector/pgvector pgvector extension documentation
+ * @see https://supabase.com/docs/guides/ai/vector-columns Supabase vector guide
+ * @see /store/doc/bridges/supabase.rst For detailed info
+ * @see /examples/rag/supabase.php Usage example
```
```suggestion
```

junaidbinfarooq replied:
Removed.

OskarStark commented on src/store/src/Bridge/Supabase/Store.php:
```diff
@@ -0,0 +1,155 @@
+        $documents = [];
+
+        foreach ($records as $record) {
+            if (
+                !isset($record['id'], $record[$this->vectorFieldName], $record['metadata'], $record['score'])
+                || !\is_string($record['id'])
+            ) {
```
one line please

OskarStark commented on src/store/tests/Bridge/Supabase/StoreTest.php:
```diff
@@ -0,0 +1,241 @@
+    {
+        $httpClient = new MockHttpClient(new MockResponse('', ['http_code' => 201]));
+        $store = $this->createStore($httpClient);
+        $doc1 = new VectorDocument(Uuid::v4(), new Vector([0.1, 0.2]), new Metadata(['a' => '1']));
+        $doc2 = new VectorDocument(Uuid::v4(), new Vector([0.3, 0.4]), new Metadata(['b' => '2']));
+
+        $store->add($doc1, $doc2);
```
```suggestion

        $store->add(
            new VectorDocument(Uuid::v4(), new Vector([0.1, 0.2]), new Metadata(['a' => '1'])),
            new VectorDocument(Uuid::v4(), new Vector([0.3, 0.4]), new Metadata(['b' => '2'])));
```

OskarStark commented on src/store/tests/Bridge/Supabase/StoreTest.php:
```diff
@@ -0,0 +1,241 @@
+    {
+        $httpClient = new MockHttpClient(new MockResponse('', ['http_code' => 201]));
+        $store = $this->createStore($httpClient);
+        $validDoc = new VectorDocument(Uuid::v4(), new Vector([0.1, 0.2]), new Metadata(['valid' => true]));
+        $invalidDoc = new VectorDocument(Uuid::v4(), new Vector([0.1]), new Metadata(['invalid' => true]));
+
+        $store->add($validDoc, $invalidDoc);
```
inline like in the other example

OskarStark commented on src/store/tests/Bridge/Supabase/StoreTest.php:
```diff
@@ -0,0 +1,241 @@
+        ];
+        $httpClient = new MockHttpClient(new JsonMockResponse($expectedResponse));
+        $store = $this->createStore($httpClient, 3);
+        $queryVector = new Vector([1.0, 2.0, 3.0]);
+        $result = $store->query($queryVector, ['max_items' => 5, 'min_score' => 0.7]);
```
```suggestion

        $result = $store->query(new Vector([1.0, 2.0, 3.0]), ['max_items' => 5, 'min_score' => 0.7]);
```

OskarStark commented on src/store/tests/Bridge/Supabase/StoreTest.php:
```diff
@@ -0,0 +1,241 @@
+        $this->assertSame(1, $httpClient->getRequestsCount());
+    }
+
+    private function createStore(?MockHttpClient $httpClient = null, ?int $vectorDimension = 2): SupabaseStore
```
```suggestion
    private function createStore(MockHttpClient $httpClient, ?int $vectorDimension = 2): SupabaseStore
```
We always provide one

junaidbinfarooq commented:
Addressed the review.
However, I'd like to mention that many of the comments I've seen in this and a few other PRs seem to be nitty-gritty IMO, and especially come at a later stage in the PR, unless there is a particular doc on coding standards in the said repo that I have missed so far.

OskarStark commented on examples/rag/supabase.php:
```diff
@@ -0,0 +1,74 @@
+    Message::ofUser('Which movie fits the theme of technology?')
+);
+
+echo "Query: Which movie fits the theme of technology?\n";
```
Remove this completely

OskarStark commented on examples/rag/supabase.php:
```diff
@@ -0,0 +1,74 @@
+
+echo "Query: Which movie fits the theme of technology?\n";
+
+try {
```
Removers try catch, only the call an echo the result, Lilien the other examples

OskarStark commented:
We try to keep the code a little bit in sync. What coding standards do you mean?
